### PR TITLE
bpo-33071: remove outdated PyPI docs

### PR DESCRIPTION
**Pull request by kojoidrissa**
https://bugs.python.org/issue33071

### CODE REVIEW

File: Doc/distutils/packageindex.rst [hunk 1]
```diff
@@ -12,242 +12,4 @@ The `Python Package Index (PyPI)`_ stores :ref:`meta-data <meta-data>`
...
-    $ python setup.py --long-description | rst2html.py > output.html
-
-:mod:`docutils` will display a warning if there's something wrong with your
-syntax.  Because PyPI applies additional checks (e.g. by passing ``--no-raw``
-to ``rst2html.py`` in the command above), being able to run the command above
-without warnings does not guarantee that PyPI will convert the content
-successfully.
-
-
-.. _Python Package Index (PyPI): https://pypi.org
```
Comment (ncoghlan) [inline Doc/distutils/packageindex.rst]: From the CI failure: this line is still needed to define the PyPI link.

**Review by eamanu — APPROVED** [review]
LGTM

Please, follow the  @ncoghlan  comments, you need to come back that line and
the documentation will be make successfully. I test it.

**Comment by ncoghlan** [comment]
@kojoidrissa I'd like to resolve this before finishing up #13092, so I went ahead and amended the update directly.

**Review by ncoghlan — APPROVED** [review]

File: Doc/distutils/packageindex.rst [hunk 2]
```diff
@@ -8,246 +8,10 @@
...
-PyPI holds a record for each (name, version) combination submitted.  The first
-user to submit information for a given name is designated the Owner of that
-name.  Changes can be submitted through the :command:`register` command or
-through the web interface.  Owners can designate other users as Owners or
-Maintainers.  Maintainers can edit the package information, but not designate
-new Owners or Maintainers.
-
-By default PyPI displays only the newest version of a given package.  The web
-interface lets one change this default behavior and manually select which
-versions to display and hide.
```
Comment (merwok) [inline Doc/distutils/packageindex.rst:36]: That’s a lot of useful information removed :-|

Comment (ncoghlan) [reply Doc/distutils/packageindex.rst:36]: Significant parts of it were incorrect information, and having it here meant that even if we fixed it now, it was likely to just become outdated again as PyPI evolves.

If there's anything that isn't sufficiently obvious from the PyPI web UI itself, then https://pypi.org/help/ would be a more appropriate location to cover it.

Comment (merwok) [reply Doc/distutils/packageindex.rst:36]: OK!

**Pull request merged**